### PR TITLE
Add return code to CancelGoal service response

### PR DESCRIPTION
**Pull request by jacobperron**
Connects to https://github.com/ros2/rcl_interfaces/issues/63

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Could you use `ERROR_NONE` in more places which currently use `0`?

**Comment by jacobperron** [comment]
> Could you use ERROR_NONE in more places which currently use 0?

I believe all the places I've left it as zero is because I'm testing if the message is zero initialized (regardless of what the name of the constant is). Does that make sense?

File: rcl_action/test/rcl_action/test_types.cpp [hunk 1]
```diff
@@ -152,6 +155,7 @@ TEST(TestActionTypes, test_init_fini_cancel_response)
   EXPECT_EQ(ret, RCL_RET_OK);
   EXPECT_EQ(cancel_response.msg.goals_canceling.size, num_goals_canceling);
   EXPECT_NE(cancel_response.msg.goals_canceling.data, nullptr);
+  EXPECT_EQ(cancel_response.msg.return_code, 0);
```
Comment (jacobperron) [inline rcl_action/test/rcl_action/test_types.cpp]: I might update this one to use the constant though.

Comment (jacobperron) [reply rcl_action/test/rcl_action/test_types.cpp]: Done 06c8c49

**Comment by jacobperron** [comment]
I've also added logic for setting the return code `GOAL_TERMINATED` in bdfb652

**Pull request merged**